lib/repl.js

```diff
@@ -436,19 +437,28 @@ function REPLServer(prompt,
       }
     }
 
-    if (errStack === '') {
-      errStack = `Thrown: ${util.inspect(e)}`;
-    }
-
     if (!self.underscoreErrAssigned) {
       self.lastError = e;
     }
 
     const top = replMap.get(self);
-    top.outputStream.write(`${errStack}\n`);
-    top.clearBufferedCommand();
-    top.lines.level = [];
-    top.displayPrompt();
+    if ((process._forceRepl || isatty(0)) &&
+        process.listenerCount('uncaughtException') !== 0) {
+      process.nextTick(() => {
+        process.emit('uncaughtException', e);
+        top.clearBufferedCommand();
```

Comment:
nit: These lines : 

```
      top.clearBufferedCommand();
      top.lines.level = [];
      top.displayPrompt();
```

are repeated in both the blocks.

Reply:
Well, I could create a function to execute them but it felt like being overkill to be so DRY here (the scope is different due to the nextTick).